base/markdown/render/terminal/render.jl

```diff
@@ -39,17 +39,37 @@ function term(io::IO, md::List, columns)
 end
 
 function term(io::IO, md::Header{1}, columns)
+    _term_header(io, md, '=', columns)
+end
+
+function term(io::IO, md::Header{2}, columns)
+    _term_header(io, md, '–', columns)
+end
+
+function term(io::IO, md::Header{3}, columns)
+    _term_header(io, md, '-', columns)
+end
+
+function _term_header(io::IO, md, char, columns)
     text = terminline(md.text)
     with_output_format(:bold, io) do io
-        print_centred(io, text, width = columns - 4margin, columns = columns)
+        print(io, " ")
+        n = print_wrapped(io, text, width = columns - 4margin; pre=" ")
+        # Note above can lose bold formatting
+        # ensure at least that the underline is bold
+        with_output_format(:bold, io) do io
+            println(io, string(char) ^ min(1 + n, columns))
```

Comment:
That would do something different. This does the following for multiline titles:

```
 This title is longer that the column width
 so underline last line
========================
```

I think using div here would get some surprising/unstable/ugly behaviour.